src/pineko/configs.py

```diff
@@ -94,16 +94,24 @@ def detect(path=None):
     pathlib.Path :
         file path
     """
-    paths = []
-
+    # If a path is provided then we only look for the pineko.toml file there.
     if path is not None:
         path = pathlib.Path(path)
-        paths.append(path)
-
-    paths.append(pathlib.Path.cwd())
-    paths.append(pathlib.Path.home())
-    paths.append(pathlib.Path(appdirs.user_config_dir()))
-    paths.append(pathlib.Path(appdirs.site_config_dir()))
+        configs_file = path / name if path.is_dir() else path
+        if configs_file.is_file():
+            return configs_file
+        else:
+            raise ValueError(
+                "Provided path is not pointing to (or does not contain) the pineko.toml file"
+            )
+    # If no path is provided we need to look after the file.
+    # We want to check pwd and all its parent folders (without their subfolders of course) up to home.
+    paths = []
+    p = pathlib.Path.cwd()
+    home = pathlib.Path.home()
+    while p != home:
+        paths.append(p)
+        p = p.parent
```

Comment:
you need to check whether `p` is contained in `home`: e.g. `/media/FK/pineko` vs. `/home/felix`

Reply:
I would not stop at home, but just at root. So:
```py
while p is not p.parent:
    ...
```

Otherwise you would have an infinite loop if you are not working in your home (for example if you are inside `/tmp`, `/media`, `/var`, ...).
And in any case, there is no reason that `/pineko.toml` or `/home/pineko.toml` should exist, so better the quitting alternative :)

Reply:
However, even easier:
```py
paths.extend(p.absolute().parents)
```